### PR TITLE
refactor: 💡 remove route-action usage for targets and sessions

### PR DESCRIPTION
**Pull request by lisbet-alvarez**
✅ Closes: https://hashicorp.atlassian.net/browse/ICU-15040

# Description

remove route action usage for targets and sessions

## How to Test

Connecting from targets list view "Connect" button && targets details view "Connect" button
1. Navigate to targets list view or details view
2. Connect to a target with address
3. Connect to a target with 1 host
    * no modal should pop up
4. Connect to a target with 2 hosts (also test quick connect)
    * modal with all hosts in host set should appear
5. Connect to a target with no address & no host set
    * error modal with retry and cancel button should appear
    * test retry and test cancel
    * Host modal should **not** appear
6. Test that "Refresh" button works as expected.

Sessions
1. Navigate to existing session details page
2. Click "End Session", and validate session is canceling
3. Navigate to sessions list page
4. Click "X" button to cancel session, and validate session is canceling
5. Test that "Refresh" button works as expected.

## Checklist

- [ ] I have added before and after screenshots for UI changes
- [ ] I have added JSON response output for API changes
- [ ] I have added steps to reproduce and test for bug fixes in the description
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: ui/desktop/.template-lintrc.js [hunk 1]
```diff
@@ -13,7 +13,6 @@ module.exports = {
     'no-bare-strings': true,
     'no-curly-component-invocation': { allow: ['app-name'] },
     'no-implicit-this': { allow: ['app-name', 'is-loading'] },
-    'no-route-action': false,
```
Comment (cameronperera) [inline ui/desktop/.template-lintrc.js:16]: 🥳

**Review by cameronperera — APPROVED** [review]
Thanks for all the hard work of removing route actions!

**Review by ZedLi — APPROVED** [review]
Fantastic work!! 🚀

**Review by calcaide — APPROVED** [review]
Let's goo!! Amazing work 🙌

**Pull request merged**